### PR TITLE
Add conditional_max_n_branches lint with i18n support

### PR DESCRIPTION
**Pull request by leynos**
## Summary
Introduce a new lint: conditional_max_n_branches, a configurable detector for complex predicate branches in if, while, and match guards. It leverages localisation helpers to render branch counts across locales and includes comprehensive CI/test coverage (unit, UI, and i18n tests). This replaces the prior two-branch concept with a flexible, locale-aware approach.

## Changes
- New lint crate: crates/conditional_max_n_branches
  - Implements a LateLintPass that inspects if/while conditions and match guards to count predicate branches and compare against a configurable max_branches (default 2).
  - Loads configuration via dylint_linting and localises messages with a shared Localizer (via common::i18n).
  - Emits diagnostics with branch and limit phrases, utilising branch_phrase to render locale-appropriate text.
  - Tests:
    - Unit tests for evaluate_condition logic and edge cases.
    - Behaviour tests scaffolding for predicate branch evaluation.
    - UI tests (fixtures) under crates/conditional_max_n_branches/ui to verify diagnostics formatting across locales.
- Library exports
  - lib.rs and module layout wired to expose the lint when the dylint-driver feature is enabled.
- i18n and localisation support
  - common now provides branch_phrase(locale, branches) to render “1 branch”/localized variants (English, Welsh, Gaelic).
  - common/src/i18n/mod.rs and lib.rs updated to re-export branch_phrase.
  - Localisation templates updated:
    - locales/en-GB/conditional_max_n_branches.ftl
    - locales/cy/conditional_max_n_branches.ftl
    - locales/gd/conditional_max_n_branches.ftl
    - cy/gd variants renaming and content adjusted accordingly
  - Documentation strings and tests updated to use branch_phrase for all locale rendering.
- Tests and test infrastructure
  - Updated common/i18n tests to rely on branch_phrase and locale-aware phrases.
  - New UI tests under crates/conditional_max_n_branches/ui with fixtures showing pass/fail cases and updated expected messages.
  - Behaviour tests under crates/conditional_max_n_branches/src/tests/behaviour.rs for predicate-branch evaluation.
- Documentation and roadmap updates
  - docs/roadmap.md reflects the new conditional_max_n_branches lint (moved from conditional_max_two_branches, now configurable).
  - docs and design notes updated to reflect localisation-driven phrasing via branch_phrase.
- UI toolchain
  - Toolchain updated to enable dylint-driver features for UI tests to exercise the lint in a realistic setting.

## Why this change
- Provides a flexible, locale-aware approach to detecting overly complex conditionals, with a configurable number of predicate branches.
- Centralises localisation through a shared branch_phrase helper to ensure consistent messaging across English, Welsh, and Gaelic, reducing duplication in Fluent templates.
- Adds end-to-end coverage (unit, UI, and localisation tests) to prevent regressions.

## How to test locally
- Build and run tests for the workspace:
  - cargo test -p conditional_max_n_branches
  - cargo test -p common (i18n tests rely on branch_phrase)
- Run dylint tests (UI):
  - cargo test -p conditional_max_n_branches --features dylint-driver
  - Inspect ui/fail_configured_limit.stderr and other fixtures for expected messages in different locales
- Manual examples:
  - If 2 branches are allowed by config, code like: if a() && b() { ... } should be allowed; a triple conjunction should warn when max_branches is 2.
  - Try with custom config max_branches to verify the limit text and note adapt accordingly (e.g., with 3 allowed, 3-branch cases should pass).

## Migration notes
- This PR modernises the previous conditional_max_two_branches concept by introducing a configurable max_branches and locale-aware phrasing. The new crate remains opt-in behind the dylint-driver feature and aligns with the existing localisation framework.

## Related links
- Localisation helper: common::i18n::branch_phrase
- Locales updated: en-GB, cy, gd for conditional_max_n_branches
- UI test fixtures and expected outputs added under crates/conditional_max_n_branches/ui

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/8b8a0bba-f8b9-4fa6-90df-cd38a8f0e628

## Summary by Sourcery

Add a new conditional_max_n_branches lint with a configurable maximum predicate branch limit and locale-aware diagnostics via a shared branch_phrase helper.

New Features:
- Introduce the conditional_max_n_branches lint crate with a configurable max_branches limit (default 2) for complex predicates in if, while, and match guards
- Provide i18n support by centralizing branch count rendering through the common::i18n::branch_phrase helper for English, Welsh, and Gaelic

Enhancements:
- Extend and re-export branch_phrase in common::i18n for consistent localization across diagnostics and UI tests
- Revise design documentation, user guide, and roadmap to reflect the new configurable lint and locale-driven phrasing

CI:
- Enable the dylint-driver feature in the UI test toolchain to run integration tests against the new lint

Documentation:
- Update design docs, users guide, and roadmap to cover conditional_max_n_branches and localization workflow

Tests:
- Add unit tests for branch counting logic and configuration parsing
- Introduce behaviour-driven tests for predicate complexity evaluation
- Add UI test fixtures across locales to verify diagnostic formatting and suggestions

### CODE REVIEW

File: crates/conditional_max_n_branches/ui/fail_if_three_branches.stderr [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   = help: Extract helper functions or simplify the �if condition� to reduce branching.
+note: the lint level is defined here
+  --> $DIR/fail_if_three_branches.rs:1:9
+   |
+LL | #![warn(conditional_max_n_branches)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+warning: 1 warning emitted
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/conditional_max_n_branches/ui/fail_if_three_branches.stderr]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.
